### ISSUE TITLE
Rule proposal: vue/prefer-vapor-mode

### ISSUE THREAD

**Issue by hadarziv-army** [vuejs/eslint-plugin-vue#2791]
**Please describe what the rule should do:**
This rule should suggest using the new `vapor` attribute on `<script setup>` blocks in Vue 3.6+.

Vapor Mode is a new feature introduced in Vue 3.6 (currently in alpha), and this rule helps developers start adopting it early by recommending `vapor` where it's appropriate. The behavior of components stays the same, but performance can be improved.

The rule should detect components using `<script setup>` and suggest adding the `vapor` attribute when possible.

**What category should the rule belong to?**
[ ] Enforces code style (layout)  
[ ] Warns about a potential error (problem)  
[x] Suggests an alternate way of doing something (suggestion)  
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- ❌ This component could use vapor mode, but doesn't -->
<script setup>
defineProps(['msg'])
</script>

<template>
  <div>{{ msg }}</div>
</template>
```
```vue
<!-- ✅ This component uses vapor mode correctly -->
<script setup vapor>
defineProps(['msg'])
</script>

<template>
  <div>{{ msg }}</div>
</template>
```

**Additional context**

[Vapor Mode](https://github.com/vuejs/core/releases/tag/v3.6.0-alpha.1) was introduced in Vue 3.6, which is currently in alpha.

This rule will become more important once Vue 3.6 is officially released, helping developers adopt the new mode and benefit from improved performance in eligible components.

**Comment by waynzh** [comment]
We have the [vue/component-api-style](https://eslint.vuejs.org/rules/component-api-style.html) rule to enforce component API style, and we could consider extending it.
